### PR TITLE
Feature/upload min fixes

### PR DESCRIPTION
**Pull request by liujoshua**

### CODE REVIEW

File: android-sdk/src/main/java/org/sagebionetworks/bridge/android/data/StudyUploadEncryptor.java [hunk 1]
```diff
@@ -29,6 +31,10 @@
 
     private final Supplier<JceKeyTransRecipientInfoGenerator> recipientInfoGeneratorSupplier;
 
+    static {
+        Security.insertProviderAt(new BouncyCastleProvider(), 1);
```
Comment (dephillipsmichael) [inline android-sdk/src/main/java/org/sagebionetworks/bridge/android/data/StudyUploadEncryptor.java]: Could you add some comments explaining what this is doing in more detail?

**Review by dephillipsmichael — APPROVED** [review]
Looks good to me.  Minor request for more code comments, but other than that you are free to merge it in.

File: android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java [hunk 2]
```diff
@@ -170,9 +171,19 @@ public Completable resendEmailVerification(@NonNull final String email) {
                     accountDAO.setStudyParticipant(
                             new StudyParticipant()
                                     .email(signIn.getEmail()));
+                }).doOnError(throwable -> {
+                    // a 412 is a successful signin
+                    if (throwable instanceof ConsentRequiredException) {
```
Comment (dephillipsmichael) [inline android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java:176]: Just curious, how is this related to the upload flow?

**Comment by liujoshua** [comment]
When I was writing junit tests and signing in, I noticed that while
unconsented,  the signIn, email, etc weren't being cached in AuthManager
like we do for a successful sign in

File: android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java (hunk 2 above)
Comment (liujoshua) [reply android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java:176]: When I was writing junit tests and signing in, I noticed that while
unconsented,  the signIn, email, etc weren't being cached in AuthManager
like we do for a successful sign in
